### ISSUE TITLE
[Feature] Reading Barcodes & QR codes to organize images and templates

### ISSUE THREAD

**Issue by Udayraj123** [Udayraj123/OMRChecker#105]
**Is your feature request related to a problem? Please describe.**
We have received this request multiple times now, reading QR codes in particular sections of the OMR, then splitting the images based on the information in the QR, and using particular template from that information. This can be extended to barcodes as well.

Note: OCR should be tracked in a different issue from this (https://github.com/Udayraj123/OMRChecker/issues/99)

**Describe the solution you'd like**
This can be implemented as a plugin. More details to follow in discussion.

**Describe alternatives you've considered**
Currently no alternatives present to scan QRs. Manual work needed.

**Additional context**
[Discord discussion](https://discord.com/channels/590134763784896514/590134763784896520/1049336842459287612)

**Comment by Udayraj123** [comment]
Linking a draft PR to mark progress: https://github.com/Udayraj123/OMRChecker/pull/116

**Comment by Udayraj123** [comment]
Found an existing product having this feature -

<img width="713" alt="image" src="https://user-images.githubusercontent.com/16881051/214732674-0c32aca6-9c31-496d-95db-d01f5c04f7ae.png">

**Comment by Udayraj123** [comment]
- Treat Barcode as a special field ✔️ 

- "fileSorting" at top level with access to read responses
    - enabled: true means it will sort per-image level
        - if there's multiple template files, only the images corresponding to current template will move
    - the "outputDirectory" from this schema will be ignored from reading images!

- The "fieldBlocks.COMMON" will contain fields common to all sets (default blocks)
- A "fieldBlocks.setMapping" at with access to  responses read from common 
    - "SET-A" etc will contain set wise fields if provided
    - This mapping should support regex as we may need to extract some pattern (roll number prefix etc)

- Later this can easily extend to reading OCR as a field 
    - https://github.com/Udayraj123/OMRChecker/issues/99
- Even the Admit card/Signature Photos can be saved and read as a field (with value = saved path) 
    - https://github.com/Udayraj123/OMRChecker/issues/73
    - Basically all fields will have access to previously calculated fields

**Comment by Udayraj123** [comment]
Crosslinking a resource on a potentially more robust 
 QR detection https://discord.com/channels/590134763784896514/590134763784896520/1164363436931690587
